Components/PetFilters.jsx

```diff
@@ -163,8 +166,36 @@ export default function PetFilters({
       })
       filters += `],`
     }
+    let profBonusFilter = []
+    let profBonusFilter1 = ``
+    let profBonusFilter2 = ``
     if (profBonus[0] > 1 || profBonus[1] < 3) {
-      filters += `profBonus_in:[${GetBonusValues(profBonus[0], profBonus[1])}],`
+      profBonusFilter1 = GetBonusValues(profBonus[0], profBonus[1])
+    }
+    if (profBonusName.length > 0) {
+      profBonusName.forEach((c, i) => {
+        profBonusFilter2 += `${c.value}`
+        if (i < profBonusName.length - 1) {
+          profBonusFilter2 += `,`
+        }
+      })
+    }
+    if ((profBonusFilter1.length > 0) && (profBonusFilter2.length > 0)) {
+      // Merge where both values exist
+      profBonusFilter = profBonusFilter1.split(",").filter(v => profBonusFilter2.split(",").indexOf(v) > -1)
+      // if none match set to an empty query
+      if (profBonusFilter.length == 0) {
+        profBonusFilter = [ 0 ]
+      }
+    }
+    else if (profBonusFilter1.length > 0) {
+      profBonusFilter = profBonusFilter1.split(",")
+    }
+    else if (profBonusFilter2.length > 0) {
+      profBonusFilter = profBonusFilter2.split(",")
```

Comment:
This is probably how I did things back when I made this code but it sounds easier to me today to just have 2 if statements instead of having the 
if both else if 1 else if 2
we can just check
if 1
if 2 add values so that if also 1 merge otherwise 2 like this:
```
if (profBonusFilter1.length > 0) {
  profBonusFilter = profBonusFilter1.split(",")
}
if (profBonusFilter2.length > 0) {
  profBonusFilter = profBonusFilter .length > 1
    ? profBonusFilter .filter(v => filter2.indexOf(v) > -1)
    : filter2
}
```

Reply:
The odd bit is where to place the merge of both filters and then knowing when both were set but incompatible to not return anything.